script/support/sample_move.rb

```diff
@@ -0,0 +1,296 @@
+# frozen_string_literal: true
+
+class StudySample < ApplicationRecord
+end
+
+# added sample class as comments are not commentable at present 21/3/2019
+class Sample
+  include Commentable
+end
+
+class Asset
+  has_many :submitted_assets
+  has_many :orders, through: :submitted_assets, as: :asset
+end
+
+def update_fluidigm_plates(fluidigm_plates)
+  fluidigm_plates.each do |plate_id|
+    puts "Updating Fluidigm plate: #{plate_id}"
+    plate = Plate.find_by id: plate_id
+
+    # rubocop:disable Rails/SkipsModelValidations
+    # correct behaviour
+    plate.touch
+    plate.save!
+    message = Messenger.find_by(target_id: plate_id)
+    message.resend
+  end
+end
+
+def find_asset_groups(sample_names)
+  hash = Hash.new { |hsh, key| hsh[key] = [] }
+  sample_names.each do |sample_name|
+    sample = Sample.find_by(name: sample_name)
+    asset_groups = sample.asset_groups
+    unless asset_groups.empty?
+      asset_groups.each { |ag| hash[ag.name] << sample_name unless Order.find_by(asset_group_id: ag.id).nil? }
+    end
+  end
+  hash
+end
+
+# rubocop:disable Metrics/PerceivedComplexity
+# rubocop:disable Metrics/MethodLength
+# rubocop:disable Metrics/AbcSize
+# rubocop:disable Metrics/CyclomaticComplexity
+# rubocop:disable Layout/LineLength
```

Comment:
If we aren't concerned about these things in scripts, would it be better to exclude files that we don't care about from Rubocop rather than disable the checkers on a file by file basis?

Reply:
I think  these should be todo's, not disables. Disable should be reserved only for false positives. I think we mostly want to fix these over time.

Also, the disabling should be scoped only to the bits of the script that is necessary, not the whole file.